### PR TITLE
Test/optimism

### PR DESCRIPTION
**Pull request by Jean-Grimal**

### CODE REVIEW

File: test/MorphoTokenOptimism.t.sol [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        // DEPLOYMENTS
+        tokenImplem = new MorphoTokenOptimism();
+        tokenProxy = new ERC1967Proxy(address(tokenImplem), hex"");
+
+        morphoOptimism = MorphoTokenOptimism(payable(address(tokenProxy)));
+        morphoOptimism.initialize(MORPHO_DAO, REMOTE_TOKEN, BRIDGE);
+    }
+
+    function testInitilizeZeroAddress(address randomAddress) public {
```
Comment (MerlinEgalite) [inline test/MorphoTokenOptimism.t.sol:34]:
```suggestion
    function testInitializeZeroAddress(address randomAddress) public {
```

File: test/MorphoTokenOptimism.t.sol [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        assertEq(morphoOptimism.bridge(), BRIDGE, "bridge");
+    }
+
+    function testMintNoBridge(address account, address to, uint256 amount) public {
+        vm.assume(account != address(0));
+        vm.assume(to != address(0));
+        vm.assume(account != BRIDGE);
+        amount = bound(amount, MIN_TEST_AMOUNT, MAX_TEST_AMOUNT);
+
+        vm.expectRevert();
```
Comment (MerlinEgalite) [inline test/MorphoTokenOptimism.t.sol:77]: I think we should expect the correct error message

File: test/MorphoTokenOptimism.t.sol [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        assertEq(morphoOptimism.balanceOf(account), amount, "balanceOf(account)");
+    }
+
+    function testBurnNoBridge(address account, address from, uint256 amount) public {
+        vm.assume(account != address(0));
+        vm.assume(from != address(0));
+        vm.assume(account != BRIDGE);
+        amount = bound(amount, MIN_TEST_AMOUNT, MAX_TEST_AMOUNT);
+
+        vm.expectRevert();
```
Comment (MerlinEgalite) [inline test/MorphoTokenOptimism.t.sol:102]: same here

File: test/MorphoTokenOptimism.t.sol [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    function testUpgrade() public {
+        address newImplem = address(new MorphoTokenOptimism());
+
+        vm.prank(MORPHO_DAO);
+        morphoOptimism.upgradeToAndCall(newImplem, hex"");
+    }
+
+    function testGetters() public view {
```
Comment (MerlinEgalite) [inline test/MorphoTokenOptimism.t.sol:66]: we should check the owner as well

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]

File: test/MorphoTokenOptimism.t.sol (hunk 2 above)
Comment (Jean-Grimal) [reply test/MorphoTokenOptimism.t.sol:78]: As the `require` takes a custom error as input, I don't know how to check this

I tried `vm.expectRevert(Error's selector)` but it doesn't seem to work

Capture d’écran 2024-10-18 à 17 10 07

**Review by MerlinEgalite — APPROVED** [review]

**Pull request merged**